main/src/ca/uwaterloo/flix/language/phase/Lexer.scala

```diff
@@ -974,21 +1005,22 @@ object Lexer {
             advance()
           }
           error = Some(TokenKind.Err(LexerError.DoubleUnderscoreInNumber(sourceLocationAtCurrent())))
+        // underscore in tail position
+        case '_' =>
+          advance()
+          return TokenKind.Err(LexerError.TrailingUnderscoreInNumber(sourceLocationAtCurrent()))
```

Comment:
I moved this so the code of the two places we parse numbers are written in the same way (it also makes the following comment tru-er)